### ISSUE TITLE
Server incorrectly setting back focus on XBL control

### ISSUE THREAD

**Issue by avernet** [orbeon/orbeon-forms#2433]
Create a form in Form Builder with a number field, name it `my-number`, publish, open the new page, click on the text field, click somewhere else on the page. Notice how the focus is set back to the number field. The last Ajax exchange is:

```xml
<xxf:event-request xmlns:xxf="http://orbeon.org/oxf/xml/xforms">
    <xxf:uuid>6d1387679912869a8d90bd27dd84b69faccb60c8</xxf:uuid>
    <xxf:sequence>3</xxf:sequence>
    <xxf:action>
        <xxf:event name="xxforms-blur" source-control-id="fr-view-wizard≡section-1-control≡xf-451≡my-number-control"></xxf:event>
    </xxf:action>
</xxf:event-request>

<xxf:event-response xmlns:xxf="http://orbeon.org/oxf/xml/xforms">
    <xxf:action>
        <xxf:control-values/>
        <xxf:focus control-id="fr-view-wizard≡section-1-control≡xf-451≡my-number-control"/>
    </xxf:action>
</xxf:event-response>
```

The client tells the server that the number lost the focus, but the server wants to set the focus back. This doesn't happen with a plain `xf:input`.

**Comment by ebruchez** [comment]
I get, when the user clicks within the input field:

- client sends `xforms-focus` to `section-1-control≡xf-386≡my-number-control` (the XBL control)
- client sends `xforms-value` to `section-1-control≡xf-386≡my-number-control≡xf-428` (the hidden input field)

On clicking outside the field:

- client sends `xxforms-blur` to `section-1-control≡xf-386≡my-number-control`
- server responds with `focus` to section-1-control≡xf-386≡my-number-control`

**Comment by ebruchez** [comment]
This issue has become a blocker with the fix for #2387, but could cause other issues as well.
